File: src/transformers/features/http/HttpIsHeadersTransformer.ts
import { HttpIsHeadersProgrammer } from "../../../programmers/http/HttpIsHeadersProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpIsHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.isHeaders",
      write: HttpIsHeadersProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/HttpIsHeadersTransformer.go
package http

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type httpIsHeadersTransformerNamespace struct{}

var HttpIsHeadersTransformer = httpIsHeadersTransformerNamespace{}

func (httpIsHeadersTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "http.isHeaders",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativehttpprogrammers.HttpIsHeadersProgrammer.Write(x)
    },
  })
}
